pages/my-dashboard.js

```diff
@@ -45,6 +52,38 @@ export default function MyDashboard(props) {
     setOpenModalWithLink({ isOpen: false, activeLink: '/' })
   }
 
+  useEffect(() => {
+    const id = setInterval(function () {
+      if (new Date() >= expires.warning && expires.active) {
+        demoContent(
+          new Date() >= expires.logout ? (
+            <SignedOut
+              closeModal={closeDemoModal}
+              onContinue={() => Router.push('./')}
+              id="SignedOut"
+              {...props.popupYouHaveBeenSignedout}
+            />
+          ) : (
+            <CountDown
+              closeModal={closeDemoModal}
+              onSignOut={() => Router.push('./')}
+              onStay={() => {
+                setExpires((t) => {
+                  return { ...t, warning: t.logout }
+                })
+                setDemoModalBody(null)
+              }}
+              id="CountDown"
+              deadline={expires.logout}
+              {...props.popupStaySignedIn}
+            />
+          )
+        )
+      } else return
+    }, 1000)
+    return () => clearInterval(id)
+  }, [expires])
```

Comment:
This might lead to some quirky behaviour since this code is run every time the expires state is updated.
It could lead to multipl intervals being set. I'm not sure it would result in any bugs since we can only have one modal but it could flicker.

I'd recommend leaving [expires] empty [] so the code only ever runs once